src/jdk.incubator.vector/share/classes/jdk/incubator/vector/X-Vector.java.template

```diff
@@ -674,91 +691,76 @@ public abstract class $abstractvectortype$ extends AbstractVector<$Boxtype$> {
                 $abstractvectortype$ neg = that.lanewise(NEG);
                 $abstractvectortype$ hi = this.lanewise(LSHL, (op == ROR) ? neg : that);
                 $abstractvectortype$ lo = this.lanewise(LSHR, (op == ROR) ? that : neg);
-                return hi.lanewise(OR, lo);
+                return m != null ? blend(hi.lanewise(OR, lo), m) : hi.lanewise(OR, lo);
             } else if (op == AND_NOT) {
                 // FIXME: Support this in the JIT.
                 that = that.lanewise(NOT);
                 op = AND;
             } else if (op == DIV) {
                 VectorMask<$Boxtype$> eqz = that.eq(($type$)0);
-                if (eqz.anyTrue()) {
-                    throw that.divZeroException();
+                if (m != null) {
```

Comment:
Checking for m!=null would cause boxing here.

Reply:
See the above solution. Thanks!

Reply:
ok.